tests-e2e/core/operation_namespace_test.go

```diff
@@ -0,0 +1,332 @@
+			err = k8s.Get(kubeSystemNamespace, k8sClient)
+			Expect(err).To(Succeed())
+
+			dummyApplicationSpec, dummyApplicationSpecString, err := createDummyApplicationData()
+			Expect(err).To(BeNil())
+
+			_, managedEnvironment, _, _, _, err := db.CreateSampleData(dbQueries)
+			Expect(err).To(BeNil())
+
+			gitopsEngineCluster, _, err := dbutil.GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID(ctx, string(kubeSystemNamespace.UID), dbQueries, log)
+			Expect(gitopsEngineCluster).ToNot(BeNil())
+			Expect(err).To(BeNil())
+
+			namespaceToTarget = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: operationNamespace}}
+			err = k8s.Get(namespaceToTarget, k8sClient)
+			Expect(err).To(Succeed())
+
+			By("creating a gitops engine instance")
+			gitopsEngineInstance := &db.GitopsEngineInstance{
+				Gitopsengineinstance_id: "test-fake-engine-instance-E2E",
+				Namespace_name:          namespaceToTarget.Name,
+				Namespace_uid:           string(namespaceToTarget.UID),
+				EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+			}
+			err = dbQueries.CreateGitopsEngineInstance(ctx, gitopsEngineInstance)
+			Expect(err).To(BeNil())
+
+			By("Create Application in Database")
+			applicationDB := &db.Application{
+				Application_id:          "test-my-application-id",
+				Name:                    applicationName,
+				Spec_field:              dummyApplicationSpecString,
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			err = dbQueries.CreateApplication(ctx, applicationDB)
+			Expect(err).To(BeNil())
+			operationDB = &db.Operation{
+				Operation_id:            "test-operation",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           db.OperationResourceType_Application,
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: "test-user",
+			}
+
+			err = dbQueries.CreateOperation(ctx, operationDB, operationDB.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			operationCR = &managedgitopsv1alpha1.Operation{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake-operation",
+					Namespace: testNamespace,
+				},
+				Spec: managedgitopsv1alpha1.OperationSpec{
+					OperationID: "test-operation",
+				},
+			}
+			err = k8sClient.Create(ctx, operationCR)
+			Expect(err).To(BeNil())
+
+			By("Verifying whether Application CR is created")
+			applicationCR := appv1.Application{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "operation",
+					Namespace: "my-user",
+				},
+			}
+
+			Eventually(func() bool {
+				isComplete, _ := dboperations.IsOperationComplete(ctx, operationDB, dbQueries)
+				fmt.Println("- Operation state result achieved : ", isComplete)
+				return isComplete
+			}, "1m", "5s").Should(BeFalse())
```

Comment:
I think this should be a `Consistently`, rather than an `Eventually`. With `Consistently`, we want to keep checking, every 5 seconds, for a minute, to make sure that the Operation is never acted upon.